### PR TITLE
Added tests for map's callbacks

### PR DESCRIPTION
**Pull request by Walfar**

### CODE REVIEW

File: app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java [hunk 1]
```diff
@@ -63,41 +86,135 @@
...
     }
 
+    @Test
+    public void displayingUserMarkerSetsPositionMarker() {
+        MainMapFragment mapFragment = fragmentTestRule.getFragment();
+        mapFragment.currentLocation = new Location("current location");
+        mapFragment.currentLocation.setLatitude(0);
+        mapFragment.currentLocation.setLongitude(0);
+
+        mapFragment.getActivity().runOnUiThread(new Runnable() {
```
Comment (jlstoupy98) [inline app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java:133]: Why do you use runOnUiThread ?

File: app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java [hunk 2]
```diff
@@ -63,41 +86,135 @@
...
+            @Override
+            public void run() {
+                mapFragment.displayNearbyMarkers();
+                if (mapFragment.googleMap != null) {
+                    mapFragment.onMarkerClick(mapFragment.activitiesMarkers.get(0));
+                }
+            }
+        });
+        assertNotNull(fAuth.getCurrentUser());
+        if (mapFragment.googleMap != null) intended(hasComponent(ActivityDescriptionActivity.class.getName()));
```
Comment (Antho1426) [inline app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java:177]: Why do you need to verify that mapFragment.googleMap is not null here?

Comment (jlstoupy98) [inline app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java:177]: Isn't the map supposed to be non-null all the time?

File: app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java [hunk 3]
```diff
@@ -63,41 +86,135 @@
 @RunWith(AndroidJUnit4.class)
 public class MainMapFragmentTest {
 
-    private UiDevice uiDevice;
+    private UiDevice uiDevice = UiDevice.getInstance(getInstrumentation());
```
Comment (Antho1426) [inline app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java:89]: Why did you change this line to use a specific instance of the uiDevice instead of the uiDevice object itself?

**Review by jlstoupy98 — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java (hunk 1 above)
Comment (Walfar) [reply app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java:133]: The callbacks methods in the map can only run on the main UI thread, so they can't be called directly in the tests. I have to make those calls run on the UI thread, and this is done by using this method !

File: app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java (hunk 2 above)
Comment (Walfar) [reply app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java:177]: Yes, actually the google map is never null when the map methods are called back. But, since Cirrus CI isn't supposed to use an internet connection, the google map would never be loaded (hence would be null), and those tests would fail. I added these checks to make sure that the tests only run under the condition that the google map is loaded. In local, these tests always run correctly, but with Cirrus, they are not doing much unfortunately.

Comment (Walfar) [reply app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java:177]: (same answer that for JL) Yes, actually the google map is never null when the map methods are called back. But, since Cirrus CI isn't supposed to use an internet connection, the google map would never be loaded (hence would be null), and those tests would fail. I added these checks to make sure that the tests only run under the condition that the google map is loaded. In local, these tests always run correctly, but with Cirrus, they are not doing much unfortunately.

File: app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java (hunk 3 above)
Comment (Walfar) [reply app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java:89]: The UIDevice was used to, for instance, click on the map. This should launch a "onMapClick" method. But, since the map was never loaded by Cirrus, this method was not called. Instead, I directly call this method in my tests (with the condition that the map is not null), which makes the UIDevice useless, so I simply removed it.

**Review by Antho1426 — APPROVED** [review]
